themes/default/content/blog/serverless-architecture-design/index.md

```diff
@@ -0,0 +1,136 @@
+
+- **Function Logic**: A serverless function should have a well-defined purpose and focus on a specific task. If a function has multiple responsibilities, consider breaking it into smaller, more focused functions. In the serverless app we deployed using Pulumi, we had two functions `date` and `weather` and it is clear what their responsibilities are.
+
+    ``` go
+    // The 'date' function that returns the current date
+	mux.HandleFunc("/api/date", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.Write([]byte(fmt.Sprintf(`{ "now": %d }`, time.Now().UnixNano()/1000000)))
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "GET")
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		weather := getWeatherData()
+
+		// Convert the weather data to JSON.
+		responseJSON, err := json.Marshal(weather)
+		if err != nil {
+			http.Error(w, "Error encoding response to JSON", http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(responseJSON)
+	})
+    ```
+
+- **Statelessness**: Serverless functions should be stateless. This encourages scalability and fault tolerance since every serverless function call should be able to function on its own, regardless of previous or subsequent calls. The recommended practise for managing data that has to be persistent beyond the span of one function execution is to store it externally in reliable data storage systems like object stores or databases. In doing so, serverless applications maintain the clear separation between function logic and stateful data management while guaranteeing data durability, consistency, and adaptability to changing workloads.
+
+- **Function triggers**: When dealing with serverless functions you should choose the Right trigger that best suits your use case. Common triggers include HTTP requests, message queues, database changes, and timers. In our application, we used an HTTP trigger for our two functions.
+
+    ``` go
+    // The 'date' function that returns the current date
+	mux.HandleFunc("/api/date", func(w http.ResponseWriter, r *http.Request) {
+        ...
+	})
+
+    // The 'weather' function that returns the weather
+	mux.HandleFunc("/api/weather", func(w http.ResponseWriter, r *http.Request) {
+        ...
+	})
+    ```
+
+    To set the trigger in your serverless function, navigate to your `function.json` configuration and set the `type` parameter in the `bindings` object.
```

Comment:
This is specific to a particular provider, is it not? We should call out the provider (Azure or AWS or GCP or whoever).

Reply:
Right, I completely understand. As the article stands now, however, it's not Azure-specific, and therefore if we are including Azure-specific pieces we should alert the reader that this is particular to Azure (and not to all serverless implementations in general).

Reply:
@AdoraNwodo This is still something that needs to be addressed (I also left another comment in the latest review).